src/main/java/com/linkedin/replica/serachEngine/commands/impl/ControllerCommand.java

```diff
@@ -0,0 +1,250 @@
+		
+		// remove double quotations of string value in JSON object
+		String s = obj.get("bytes").toString().replaceAll("\"", "");
+		String packageName = obj.get("packageName").toString().replaceAll("\"", "");
+		String fileName = obj.get("fileName").toString().replaceAll("\"", "");
+//		String commandConfigPropKey = obj.get("configPropKey").toString().replaceAll("\"", "");
+
+		// decode to bytes to write file
+		byte[] bytes = Base64.getDecoder().decode(s);
+		
+		// write file to .class folder in commands/impl package
+		writeFile(fileName, packageName,  bytes);
+		
+		// update configuration file
+		// TODO
+	}
+	
+	/**
+	 * freeze the independent app  i.e. stop accepting new requests and release all resources from pools
+	 * @param val
+	 */
+	public static void freeze(Object val){
+		// TODO
+	}
+	
+	/**
+	 * continue accepting new requests and ask pools to acquire resources again
+	 * @param val
+	 */
+	public static void resume(Object val){
+		// TODO 
+	}
+	
+	/**
+	 * set the type of errors that must be logged
+	 * @param val
+	 */
+	public static void setErrorReportingLevel(Object val){
+		if(val == null || ! ((JsonPrimitive) val).isNumber())
+			throw new SearchException(String.format("Invalid parameters : %s. expected an integer representing logging level.", val ));
+		
+		int loggingLevel = ((JsonPrimitive) val).getAsInt();
+		config.setAppControllerProp("app.error_reporting_level", loggingLevel+"");
+		// TODO 
+	}
+
+	/**
+	 * convert an embedded JSON object to JSONObject
+	 * @param val
+	 * @return
+	 */
+	private static JsonObject convert(Object val){
+		Gson gson = new Gson();
+		return gson.fromJson(val.toString(), JsonObject.class);
+	}
+	
+	private static void validate(JsonObject obj, String... args){
+		for(String arg : args){
+			if(obj.get(arg) == null )
+				throw new SearchException(String.format("Invalid parameters : %s. expected : %s", obj.keySet().toString(), Arrays.toString(args)));
+		}
+	}
+	
+	private static void writeFile(String fileName, String packageName, byte[] bytes) throws IOException{
+		// get .class folder path from configuration
+		String folderPath = config.getAppConfigProp("app.classes.path");
+		// get full path of new file
+		Path path = Paths.get(folderPath + "/" + packageName + "/" + fileName+".class");
+		// delete file if exist to avoid throwing FileAlreadyExistsException  
+		Files.deleteIfExists(path);
+		// write file
+		Files.write(path, bytes, new OpenOption[]{StandardOpenOption.CREATE_NEW});
+	}
+	
+	private static void deleteFile(String fileName, String packageName) throws IOException{
+		// get .class folder path from configuration
+		String folderPath = config.getAppConfigProp("app.classes.path");
+		// get full path of new file
+		Path path = Paths.get(folderPath + "/" + packageName + "/" + fileName+".class");
+		// delete file if exist to avoid throwing FileAlreadyExistsException  
+		Files.deleteIfExists(path);
+	}
+	
+	public static void main(String[] args) throws IOException {
```

Comment:
Do we need the main method here? Maybe we can move its implementation to some unittest?